presto-raptor/src/main/java/com/facebook/presto/raptor/util/DatabaseUtil.java

```diff
@@ -134,16 +140,26 @@ public static boolean isSyntaxOrAccessError(Exception e)
         return sqlCodeStartsWith(e, "42");
     }
 
+    public static boolean isTransactionCacheFullError(Exception e)
+    {
+        return sqlErrorCodeMatches(e, ER_TRANS_CACHE_FULL);
+    }
+
+    private static boolean sqlErrorCodeMatches(Exception e, int errorCode)
+    {
+        return Throwables.getCausalChain(e).stream().anyMatch(t -> (t instanceof SQLException) && ((SQLException) t).getErrorCode() == errorCode);
```

Comment:
When using streams, you want to do the logic functionally, instead of putting a complex expression in a lambda:
```java
return Throwables.getCausalChain(e).stream()
        .filter(SQLException.class::isInstance)
        .map(SQLException::getErrorCode)
        .anyMatch(errorCode);